components/Common/ThemeToggle/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { MoonIcon, SunIcon } from '@heroicons/react/24/outline';
+import { AccessibleIcon } from '@radix-ui/react-accessible-icon';
+import { useTheme } from 'next-themes';
+import type { FC } from 'react';
+import { useIntl } from 'react-intl';
+
+import styles from './index.module.css';
+
+const ThemeToggle: FC = () => {
+  const intl = useIntl();
+  const { theme, setTheme } = useTheme();
```

Comment:
This Component should be generic and not wired up to `useTheme`. 

I would argue that this Component receives one prop:

`{ onClick: () => void `}

Reply:
I don't really agree with you, because the component is designed to switch between themes. What you're proposing would be for a "feature toggle" component and you'd have props like that:
```ts
{
onClick: () => void ;
value: number
feature: ReactNode[] | string[]
}

Reply:
Im not proposing a feature toggle. And the component should be generic.

Components in general should be as simple as possible.

It doesn't need any "value" logic.

It's still a theme toggle because it's contextually aware of the dark theme/light theme through Tailwind and has an onClick handler.

Reply:
Also in no moment I mentioned any of the props you're shown on your comment.

Reply:
I got it !